synapse/handlers/sso.py

```diff
@@ -24,13 +24,14 @@
     Iterable,
     List,
     Mapping,
+    NoReturn,
     Optional,
     Set,
 )
 from urllib.parse import urlencode
 
 import attr
-from typing_extensions import NoReturn, Protocol
+from typing_extensions import Protocol
```

Comment:
Protocol was added in 3.8, see https://docs.python.org/3/library/typing.html?highlight=typing%20protocol#typing.Protocol, so wecan import this from `typing` now.

Reply:
(I think ignore this comment and merge this as-is. We'll still need typing extensions for things like ParamSpec.)

Reply:
I'm surprised that it didn't do this automatically TBH...